torch/_inductor/lowering.py

```diff
@@ -3898,7 +3898,7 @@ def mul(a, b):
 #   integer inputs and true division for floating and complex inputs.
 @register_lowering([prims.div], broadcast=True)
 def div_prim(a, b):
-    is_integral = is_boolean_type(a) or is_integer_type(a)
+    is_integral = all(is_boolean_type(x) or is_integer_type(x) for x in [a, b])
```

Comment:
Why does type promotion not promote scalar values?

Reply:
yeah I thought it would, but it doesn't seem to in this case :/ Didn't look more into it since this fix is fairly reasonable.

Reply:
I guess `fmod` will need the same fix then, since it uses the same `is_integral` test.

Reply:
I can't seem to hit the same codepath with `fmod`, perhaps because it doesn't take in a scalar as an input.